neuralop/training/training_state.py

```diff
@@ -71,6 +76,10 @@ def save_training_state(save_dir: Union[str, Path], save_name: str,
     save_name : str
         name of model to load
     """
+    # if running in DDP, only save state on rank 0
+    if dist.is_initialized():
```

Comment:
We should not be handling this, at least not here -- the user should be responsible for calling this only on the rank(s) they want.

Reply:
got it